cvelib/cli.py

```diff
@@ -425,19 +425,27 @@ def show_cve(ctx: click.Context, show_record: bool, print_raw: bool, cve_id: str
     cve_api = ctx.obj.cve_api
 
     cve_id_data = cve_api.show_cve_id(cve_id=cve_id)
-    cve_record_data = None
+    cve_record_data = {}
     if show_record:
-        cve_record_data = cve_api.show_cve_record(cve_id=cve_id)
+        try:
+            cve_record_data = cve_api.show_cve_record(cve_id=cve_id)
+        except requests.exceptions.HTTPError as exc:
+            error_msg = exc.response.json()["error"]
+            if exc.response.status_code != 404 or error_msg != cve_api.Errors.RECORD_DOES_NOT_EXIST:
+                raise exc
 
     if print_raw:
         print_json_data(cve_id_data)
+        if show_record:
+            print_json_data(cve_record_data)
     else:
         print_cve_id(cve_id_data)
-        if cve_record_data:
+        if show_record:
             click.secho("-----", bold=True)
-
-    if cve_record_data:
-        print_json_data(cve_record_data)
+            if cve_record_data:
+                print_json_data(cve_record_data)
+            else:
+                click.echo("CVE record does not exists.")
```

Comment:
This seems a little messy, can we just unindent one level like:
```suggestion
    if print_raw:
        print_json_data(cve_id_data)
    else:
        print_cve_id(cve_id_data)
    if show_record:
        click.secho("-----", bold=True)
        if cve_record_data:
            print_json_data(cve_record_data)
        else:
            click.echo("CVE record does not exist.")
```

Reply:
That's slightly different in that you're no longer printing just JSON in the raw output and printing the error text message instead of the empty object. No other command really shows output of two different responses in the raw view, and combining them into one JSON object feels weird too, so I figured I'd just list them one after another with a missing record represented as `{}`. Maybe the delimiter still makes sense to add so that there is _something_ to split on instead of reading until end of first and then expecting the second.

On second thought, as I was about to submit this, I think I will just spit it out as a single array of two items so that it is still possible to parse it into valid JSON and just except the user to read the first element as the first response and second element as second. Perhaps less unexpected than the alternative I described above :man_shrugging:

Reply:
Ah, that makes sense - I hadn't quite understood what the differences were in the two cases. So the error message you have here works for the human-readable case, and a JSON array with two items like [{cve_id_data}, {cve_record_data or empty}] sounds good for the machine-readable case. LGTM.